packages/firestore/src/remote/remote_event.ts

```diff
@@ -70,6 +70,70 @@ export class RemoteEvent {
       resumeToken: emptyByteString()
     };
   }
+
+  /**
+   * Strips out mapping changes that aren't actually changes. That is, if the document already
+   * existed in the target, and is being added in the target, and this is not a reset, we can
+   * skip doing the work to associate the document with the target because it has already been done.
+   */
+  filterUpdatesFromTargetChange(
```

Comment:
I am not a huge fan of this method.

1) A `filter` in JS usually returns a modified list that had the filter applied to it. The original list is not modified.

2) This method (and it's location) suggests that it modifies the encompassing RemoteEvent. It doesn't seem to do this, and could be implemented as a function outside of this class.

3) It's difficult to tell what is actually being updated here. The inline variable make the code much easier to read, but they make it hard to answer the question what `result` and `update` actually are.

Reply:
I understand that this is the ported version of your original PR. I am still a little concerned though since this method changes internal and external state at the same time, while its naming indicates something else entirely.

I know that this is far from what iOS does, but I would prefer something like:


```
filterKnownDocuments(targetId:TargetID, knownDocuments:..) : RemoteEvent {
   ... this reads internal state, but doesn't mutatate it.
}
```
or:

```
removeKnownDocuments(targetId:TargetID, knownDocuments:..) : void {
   ... this reads internal state and also mutatates it.
}
```

In the interim, and to keep the port close, we could also make this a bit better by making it a static method.

Reply:
As discussed in person, kept the method non-static, but I switched the parameter to be `targetId` rather than `targetChange`, and added the requisite null check inside the method.